src/types/types.ts

```diff
@@ -693,7 +693,16 @@ export interface EdgeDataDump {
   }
 }
 
+export interface EdgeAddress {
+  addressType: string // 'publicAddress' | 'segwitAddress' | 'legacyAddress' | 'fooAddress'
+  publicAddress: string
+  nativeBalance?: string
+}
+
 export interface EdgeFreshAddress {
+  allAddresses?: EdgeAddress[]
```

Comment:
So we need to do the reverse translation:

In the `EdgeCurrencyWallet.getFreshAddress` method, we should call `EdgeCurrencyEngine.getAddresses` if it exists, and then down-convert into this older `EdgeFreshAddress` format. That way, engines can stop implementing `getReceiveAddress`, but we can still provide the same core API to the GUI.